feature-ledger-impl/src/main/java/io/novafoundation/nova/feature_ledger_impl/presentation/account/common/selectLedger/stateMachine/states/SelectDiscoveryModeState.kt

```diff
@@ -0,0 +1,18 @@
+package io.novafoundation.nova.feature_ledger_impl.presentation.account.common.selectLedger.stateMachine.states
+
+import io.novafoundation.nova.common.utils.stateMachine.StateMachine
+import io.novafoundation.nova.feature_ledger_api.sdk.discovery.DiscoveryMethod
+import io.novafoundation.nova.feature_ledger_impl.presentation.account.common.selectLedger.stateMachine.SelectLedgerEvent
+import io.novafoundation.nova.feature_ledger_impl.presentation.account.common.selectLedger.stateMachine.SideEffect
+
+class SelectDiscoveryModeState : SelectLedgerState() {
+
+    override suspend fun StateMachine.Transition<SelectLedgerState, SideEffect>.performTransition(event: SelectLedgerEvent) {
+        if (event is SelectLedgerEvent.DiscoveryMethodSelected) {
+            when (event.discoveryMethod) {
+                DiscoveryMethod.BLE, DiscoveryMethod.ALL -> WaitingForPermissionsState(event.discoveryMethod)
+                DiscoveryMethod.USB -> startDiscovery(event.discoveryMethod)
```

Comment:
Again, we should not hardcode that a particular discovery method requires permissions and others - don't